scripts/Muon/GUI/Common/utilities/xml_utils.py

```diff
@@ -116,24 +119,43 @@ def load_grouping_from_XML(filename):
         default = ''
 
     group_names, group_ids, periods = _get_groups_from_XML(root)
-    pair_names, pair_groups, pair_alphas = _get_pairs_from_XML(root)
-    diff_names, diff_groups = _get_diffs_from_XML(root)
+    pair_names, pair_groups, pair_alphas, pair_periods = _get_pairs_from_XML(root)
+    diff_names, diff_groups, diff_periods = _get_diffs_from_XML(root)
     groups, diffs, pairs = [], [], []
 
     for i, group_name in enumerate(group_names):
-        period = periods[i] if periods and i<len(periods) else [1]
+        period = periods[i] if periods and i < len(periods) else [1]
         groups += [MuonGroup(group_name=group_name,
                              detector_ids=group_ids[i], periods=period)]
 
     for i, pair_name in enumerate(pair_names):
+        if pair_periods:
+            pair_periods_converted = []
+            for p in pair_periods[i]:
+                try:
+                    pair_periods_converted += [int(p)]
+                except ValueError:
+                    continue
+        else:
+            pair_periods_converted = [1]
+
         pairs += [MuonPair(pair_name=pair_name,
                            forward_group_name=pair_groups[i][0],
                            backward_group_name=pair_groups[i][1],
-                           alpha=pair_alphas[i])]
+                           alpha=pair_alphas[i],
+                           periods=pair_periods_converted)]
 
     for i, diff_name in enumerate(diff_names):
-        diffs +=[MuonDiff(diff_name,diff_groups[i][0],
-                          diff_groups[i][1], diff_groups[i][2], diff_groups[i][3])]
+        if diff_periods:
+            diff_periods_converted = []
+            for p in diff_periods[i]:
+                try:
+                    diff_periods_converted += [int(p)]
+                except ValueError:
+                    continue
```

Comment:
Can these two similar for loops on the pair_periods & diff_periods be extracted into a reusable function?